apps/user-profile/src/components/modules/OnboardingLandingPage/OnboardingLandingPage.js

```diff
@@ -1,272 +1,385 @@
+    const profileId = userData?.profile?.id;
 
-    const onCancelClick = (e) => {
-        router.push({
-            pathname: '/',
-        }, '/');
-        e.preventDefault()
-    }
+    const requestBody = {
+      data: {
+        displayName: `${person.firstName} ${person.lastName}`,
+        profilePictureUrl: uploadedProfilePicture || profilePicture,
+      },
+    };
+    profileId &&
+      axios
+        .put(
+          `${process.env.NEXT_PUBLIC_STRAPI_URL}/profiles/${profileId}/`,
+          requestBody,
+          { withCredentials: true }
+        )
+        .then(() => {
+          console.log('Profile picture updated successfully');
+          router.push(
+            {
+              pathname: '/users/me',
+              query: { onboarding: true },
+            },
+            '/users/me'
+          );
+        })
+        .catch((error) => {
+          console.error('Error updating profile picture:', error);
+        });
+    e.preventDefault();
+  };
 
-    return (
-        <>
-            <PageContainer>
-                <BannerContainer>
-                    <img src={Breadcrumb} />
-                </BannerContainer>
-                <OnboardingContainer>
-                    <ProfileContainer>
-                        <ProfileHeader>
-                            <Typography type="p">PROFILE</Typography>
-                            <Typography type="h2">Let's Create Your Profile</Typography>
-                        </ProfileHeader>
-                        <Typography>A completed profile will help us match you with opportunities</Typography>
-                        <UploadProfilePicture width={120} height={120}></UploadProfilePicture>
-                    </ProfileContainer>
-                    <FormContainer>
-                        <Typography>* Indicates a required field</Typography>
-                        <FormFields name="myForm"
-                        >
-                            <InputField
-                                error="Please fill in your First Name"
-                                label="First Name"
-                                name="First Name"
-                                onChange={onFirstNameChange}
-                                placeholder="John"
-                                touched={person.firstNameTouched ? !formValidation.isFirstNameValid : false}
-                                required
-                            />
-                            <InputField
-                                error="Please fill in your Last Name"
-                                label="Last Name"
-                                name="Last Name"
-                                onChange={onLastNameChange}
-                                placeholder="Doe"
-                                touched={person.lastNameTouched ? !formValidation.isLastNameValid : false}
-                                required
-                            />
+  const onCancelClick = (e) => {
```

Comment:
For my understanding, this redirects the user to the home page?

Reply:
Yes